Add document for Entra ID configuration (#436)
pulibrary/pul-it-handbook: services/entra-id.md

   <https://your-production-app.princeton.edu/users/auth/entra_id/callback>

   
1. Add the redirect URI under the Web platform type for a traditional server-rendered Rails application. Microsoft lists Ruby server-side web applications under the Web redirect URI configuration.

   <img width="1066" height="688" alt="Screenshot 2026-07-02 at 2 18 38 PM" src="https://github.com/user-attachments/assets/cfcf6a91-ea5d-4964-8409-1f1bb0420470" />
   
1. Create a client secret for the app registration.
   <img width="1446" height="705" alt="Screenshot 2026-07-06 at 10 40 15 AM" src="https://github.com/user-attachments/assets/5d4426b8-7e57-4dad-adba-95c9d35e3a72" />

1. Save the following values for the Rails application in last pass You will only see the client-secret-value once:

   ENTRA_CLIENT_ID=<application-client-id>
   ENTRA_CLIENT_SECRET=<client-secret-value> # created from `client credentials` in Step. 5
   ENTRA_TENANT_ID=2ff60116-7431-425d-b5af-077d7791bda4 # this is the global PUL ID

   **Note**: for development you should save the information, but as you will be the only user it does not need to be shared

> 1. Add the rest of your team to the Owners of the application (unless this is a devlopemtn key)

Reviewer: ```suggestion
1. Add the rest of your team to the Owners of the application (unless this is a development key)
```

   <img width="1458" height="909" alt="Screenshot 2026-07-02 at 2 22 58 PM" src="https://github.com/user-attachments/assets/cfc41c32-ad63-4dad-ad92-c4fd1d8810fe" />

1. Add attributes you need to have access to.  (In theory you can add groups, but that does not seem to be working)

>    This is optional, you could also just utilize the information from info.  That said no uid  like acb123 is present in the infor and the email is your alias email. 

Reviewer: ```suggestion
   This is optional, you could also just utilize the information from info.  That said no uid like `abc123` is present in the info, and the email is your alias email. 
```

   <img width="1461" height="837" alt="Screenshot 2026-07-06 at 10 15 09 AM" src="https://github.com/user-attachments/assets/dcfd19fe-f5a2-4f50-ab24-ceaf2f3aa311" />

## Rubyapp Integration

1. Update your Gemfile to include Entra ID:

   ```
    # Single sign on
    gem "omniauth-entra-id"
   ```

1. Install the gems:

    bundle install

1. Information from entra will be returned in the request
   ```
   access_token.extra.raw_info.<key your configured>
   ```

> 1. The UID can be found by pulling the uniue name

Reviewer: ```suggestion
1. The UID can be found by pulling the unique name
```

   ```
   # full email abc123@princeton.edu
   access_token.extra.raw_info.unique_name

   # only uid abc123
   access_token.extra.raw_info.unique_name.split('@princeton.edu').first
   ```

### Hanami
1. Add a setting to the app/settings.rb
   ```
   setting :entra_client_id, default: '', constructor: Types::String
   setting :entra_client_secret, default: '', constructor: Types::String
   ```
1. Add configuration to app/application.rb
   ```
   require 'omniauth-entra-id'

   ...
   # inside class App < Hanami::App
     config.middleware.use OmniAuth::Builder do
        provider :entra_id, client_id: Hanami.app.settings.entra_client_id, client_secret: Hanami.app.settings.entra_client_secret
     end
   ...
   ```
1. Export the secrets to your environment
   export ENTRA_CLIENT_SECRET=<client-secret-value>
   export ENTRA_CLIENT_ID=<application-client-id>

1. Add an additional route for the call back.  Note the URL will be `/auth/entra_id/callback` as specified by the gem

>    In config routes add (the to can change to be any location you wish the entra callback to be)

Reviewer: ```suggestion
   In config routes add (the `to` setting can change to be any location you wish the entra callback to be)
```

   ```
   get "/auth/entra_id/callback", to: "auth.entra"
   ```
1. Add an action to handle the callback similar to CAS

>    **Example code below does not handle errors and assumed the method from_entra_id parse the authorization hash appropriately**

Reviewer: ```suggestion
   **Example code below does not handle errors and assumes the method from_entra_id can parse the authorization hash appropriately**
```
